src/Evaluator.ml

```diff
@@ -20,6 +22,7 @@ let rec free_vars_m Location.{ data = term; _ } =
   | Letbox { idm; boxed; body } ->
       Set.union (free_vars_m boxed)
         (Set.diff (free_vars_m body) (Set.singleton (module Id.M) idm))
+  | Match _ -> Set.empty (module Id.M)
```

Comment:
We need to walk the expression we match on and the branches and take their union.

Reply:
Done